scripts/react_native_pods.rb

```diff
@@ -111,11 +116,15 @@ def flipper_post_install(installer)
 
 # Pre Install processing for Native Modules
 def codegen_pre_install(installer, options={})
-  # Path to React Native
-  prefix = options[:path] ||= "../node_modules/react-native"
+  # Path to React Native relative to where `pod install` was invoked. This can
+  # sometimes be inside the `ios` folder (i.e. `pod install `), or outside
+  # (e.g. `pod install --project_directory=ios`).
+  prefix = options[:path] ||= resolve_module "react-native"
```

Comment:
To keep it the same with `use_react_native`, we should probably just stick to `options[:path]` and pass it like here:

https://github.com/facebook/react-native/blob/355239b99125ffc7d17009a1c71e2fc42f7c104a/template/ios/Podfile#L7-L9

Reply:
`reactNativePath` can be programatically set in the `react-native.config.js` and might be not what this resolves to in `use_react_native!`.

Example: https://github.com/facebook/react-native/blob/master/react-native.config.js#L32

Set to `.` for React Native root, this will not work inside React Native repository as there's no module to `require.resolve` to.

Reply:
So, `codegen_pre_install(:reactNativePath => config[:reactNativePath])`, would do?

Reply:
BTW, if we make `codegen_pre_install` to use `config[:reactNativePath]` which is an absolute path to React Native location, then, path to codegen would always be correct, since it's absolute and independent of location of a project.

Reply:
Technically, the assumption of `react-native-codegen` living side by side `react-native` can be false in the following scenario:
```
node_modules/react-native-codegen (0.0.7)
packageA/node_modules/react-native (0.64.0-rc.0)
packageB/node_modules/react-native (0.64.0-rc.1)
```
In other words, two React Native packages sharing same React Native codegen version. 

In that case, it may be worth to leave `resolve_module` just for `react-native-codegen` or... we make sure that `react-native-codegen` is always versioned together with the React Native (same version as the RC, with patch bump). 

This is the idea going forward, but at the time of writing, the aforementioned scenario is technically possible.

Reply:
Not quite sure I understand what you mean here… The root of this issue is that we cannot use the same path for both `use_react_native!` and `codegen_pre_install`. `use_react_native!` needs paths that are relative to the `Podfile` because CocoaPods will be resolving modules later. `codegen_pre_install`, on the other hand, needs the paths to be relative to the current working directory because it is invoked right away.

Reply:
Sorry, didn't see your replies until after I had posted this. The scenario you mentioned in the last post is a very real one for us. I'd rather leave it like this.